### DISCUSSION TITLE
UPBGE 0.3 - development idea - mouse actuator

### DISCUSSION THREAD

**Discussion by SwanTheHunt** [UPBGE/upbge#1425] [Ideas/Features for UPBGE 0.3+]
**A valid idea to avoid using scripts**

The "Mouse Sensor" could have more functionality for the "Mouse Sensor Movement" or "Mouse Over"
Add motion functionality, not just rotation, to the "Mouse Actuator". So it would look like this:

- [ ] Mouse Sensor > Controller > Mouse Actuator > **Mode [Visibility / Look / Motion]** 

The "Motion mode" would control the object's position in XYZ (but it could be restricted) and not just its rotation like the "Look". Obviously this is a mode that would work in conjunction with the Camera perspective. Example: Moving folders from one place to another on the desktop

**Comment by youle31** [comment]
I moved this thread to discussions as it is not a bug report

**Comment by BluePrintRandom** [comment]
we can use 

object_to_move.worldPosition = mouse.hitPosition

or like this - the object will always be the same distance away even if the hitpoint is far
v2 = own.scene.active_camera.getVectTo(mouse.hitPosition)
object_to_move.worldPosition = own.scene.active_camera.worldPosition + (v2[1] * distance)
